km/km_filesys.c

```diff
@@ -75,6 +77,56 @@ static int km_fs_g2h_readlink(const char* name, char* buf, size_t bufsz);
 km_fd_dup_data_t dup_data;
 pthread_mutex_t dup_data_mtx;
 
+/*
+ * Lightweight snap start - wait for connection on the snap_listen_sock before restoring the snapshot.
+ */
+static const_string_t KM_SNAP_LISTEN_PORT = "SNAP_LISTEN_PORT";
+
+static int snap_listen_sock = -1;
+static km_fd_socket_t* snap_listen_sockinfo;
+static int km_internal_fd(int fd, int km_fd);
+
+void light_snap_listen(void)
+{
+   char* port;
+   if ((port = getenv(KM_SNAP_LISTEN_PORT)) == NULL) {
+      return;
+   }
+   snap_listen_sock = socket(AF_INET, SOCK_STREAM, 0);
+
+   snap_listen_sockinfo = malloc(sizeof(km_fd_socket_t));
+   *snap_listen_sockinfo = (km_fd_socket_t){.domain = AF_INET, .type = SOCK_STREAM, .protocol = 0};
+
+   struct sockaddr_in sa_serv = {.sin_family = AF_INET, .sin_addr.s_addr = INADDR_ANY};
```

Comment:
See this document: https://tldp.org/HOWTO/Linux+IPv6-HOWTO/ch23s01.html.
In particular, see the example in section `1.4.1.  'Daytime' Server Code'`

Note how `getaddrinfo` is used to get a list of constructed socket addresses, both IPv4 and IPv6, to be used by `bind`. This is the way.

Reply:
I tried that example program - the end result boils down to the same thing. Granted your way is the better way but it ain't my problem ATM.